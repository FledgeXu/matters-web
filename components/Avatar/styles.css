img {
  border-radius: 50%;
}

/* 20px */
.xsmall {
  width: 1.25rem;
  height: 1.25rem;
}

/* 30px */
.small {
  width: 1.875rem;
  height: 1.875rem;
}

/* 40px */
.default {
  width: 2.5rem;
  height: 2.5rem;
}

/* 70px */
.large {
  width: 4.375rem;
  height: 4.375rem;
}

/* 140px */
.xlarge {
  width: 8.75rem;
  height: 8.75rem;
}
