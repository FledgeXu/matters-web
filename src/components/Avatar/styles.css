.avatar {
  /* may under a flex layout, keep it unshrinkable */
  flex-shrink: 0;
  display: inline-block;

  /* 40px */
  width: 2.5rem;
  height: 2.5rem;

  border-radius: 50%;
  background-color: var(--color-grey-lighter);
}

/* 16px */
.xs {
  width: 1rem;
  height: 1rem;
}

/* 20px */
.sm {
  width: 1.25rem;
  height: 1.25rem;
}

/* 24px */
.md {
  width: 1.5rem;
  height: 1.5rem;
}

/* 32px */
.lg {
  width: 2rem;
  height: 2rem;
}

/* 48px */
.xl {
  width: 3rem;
  height: 3rem;
}

/* 72px */
.xxl {
  width: 4.5rem;
  height: 4.5rem;
}
