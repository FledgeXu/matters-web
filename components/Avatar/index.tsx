import classNames from 'classnames'
import gql from 'graphql-tag'

import ICON_AVATAR_DEFAULT from '~/static/icons/avatar-default.svg'

import { AvatarUser } from './__generated__/AvatarUser'
import styles from './styles.css'

type AvatarSize =
  | 'xxxsmall'
  | 'xxsmall'
  | 'xsmall'
  | 'small'
  | 'default'
  | 'large'
  | 'xlarge'

const fragments = {
  user: gql`
    fragment AvatarUser on User {
      avatar
    }
  `
}

export const Avatar = ({
  user,
  size = 'default'
}: {
  size?: AvatarSize
  user?: AvatarUser
}) => {
  const src = (user && user.avatar) || ICON_AVATAR_DEFAULT
  const avatarClasses = classNames({
    avatar: true,
    [size]: true
  })

  return (
    <>
      <div
        className={avatarClasses}
        style={{ backgroundImage: `url(${src})` }}
        aria-hidden="true"
      />
      <style jsx>{styles}</style>
    </>
  )
}

Avatar.fragments = fragments
