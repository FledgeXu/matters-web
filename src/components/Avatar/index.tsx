import classNames from 'classnames'
import { gql } from '@apollo/client'

import { Img } from '~/components'

import ICON_AVATAR_DEFAULT from '@/public/static/icons/avatar-default.svg'
import IMAGE_CIVIC_LIKER_RING from '@/public/static/icons/civic-liker-ring.svg'

import styles from './styles.css'

import { AvatarUser } from './__generated__/AvatarUser'

export type AvatarSize = 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl'

export interface AvatarProps {
  user?: AvatarUser
  size?: AvatarSize
  src?: string
  inEditor?: boolean
}

const fragments = {
  user: gql`
    fragment AvatarUser on User {
      avatar
      liker {
        civicLiker
      }
    }
  `,
}

export const Avatar = (props: AvatarProps) => {
  const { user, size = 'default', src, inEditor } = props
  const source = src || user?.avatar || ICON_AVATAR_DEFAULT
  const isFallback =
    (!src && !user?.avatar) || source.indexOf('data:image') >= 0
  const isCivicLiker = user?.liker.civicLiker
  const avatarClasses = classNames({
    avatar: true,
    [size]: true,
    'civic-liker': isCivicLiker,
  })

  return (
    <div className={avatarClasses}>
      <Img url={source} size="144w" disabled={isFallback || inEditor} />

      {isCivicLiker && <span className="ring" />}

      <style jsx>{styles}</style>

      <style jsx>{`
        .ring {
          background-image: url(${IMAGE_CIVIC_LIKER_RING});
        }
      `}</style>
    </div>
  )
}

Avatar.fragments = fragments
