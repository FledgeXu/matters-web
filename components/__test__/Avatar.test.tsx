import { render } from '@testing-library/react'
import React from 'react'

import { Avatar } from '../Avatar'

test('Avatar default style match snapshot', () => {
  const { container } = render(<Avatar />)
  expect(container.firstChild).toMatchSnapshot()
})
