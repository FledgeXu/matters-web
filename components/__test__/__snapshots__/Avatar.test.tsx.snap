// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar default style match snapshot 1`] = `
<div
  class="jsx-undefined avatar default"
  style="background-image: url(avatar-default.svg);"
/>
`;
